doc/man3/EVP_PKEY_fromdata.pod

```diff
@@ -50,9 +61,156 @@ EVP_PKEY_fromdata() return 1 for success and 0 or a negative value for
+
+These examples are very terse for the sake of staying on topic, which
+are the EVP_PKEY_fromdata() set of functions.  In real applications,
+BIGNUMs would be handled, and converted to byte arrays with
+BN_bn2nativepad(), but that's off topic here.
+
+=begin comment
+
+TODO Write a set of cookbook documents and link to them.
+
+=end comment
+
+=head2 Creating an RSA keypair using raw key data
+
+ #include <openssl/evp.h>
+
+ /*
+  * These are extremely small to make this example simple.  A real
+  * and secure application will not use such small numbers.  A real
+  * and secure application is expected to use BIGNUMs, and to build
+  * this array dynamically.
+  */
+ const unsigned long rsa_n = 0xbc747fc5;
+ const unsigned long rsa_e = 0x10001;
+ const unsigned long rsa_d = 0x7b133399;
+ const OSSL_PARAM[] = {
+     OSSL_PARAM_ulong("n", &rsa_n),
+     OSSL_PARAM_ulong("e", &rsa_e),
+     OSSL_PARAM_ulong("d", &rsa_d),
+     OSSL_PARAM_END
+ };
+     
+ int main()
+ {
+     EVP_PKEY_CTX *ctx = EVP_PKEY_CTX_new_from_name(NULL, "RSA", NULL);
+     EVP_PKEY *pkey = NULL;
+
+     if (ctx == NULL
+         || !EVP_PKEY_key_fromdata_init(ctx)
+         || !EVP_PKEY_fromdata(ctx, &pkey, params))
+         exit(1);
+
+     /* Do what you want with |pkey| */
+ }
+
+=head2 Creating an ECC keypair using raw key data
+
+ #include <openssl/evp.h>
+
+ /*
+  * These arrays represent large numbers, big endian organization.
+  * In a real application, these would probably be bignums that get
+  * converted to the native integer organization with BN_bn2nativepad().
+  * We're not doing that here, since this is not an of BIGNUM
+  * functionality, but an example of EVP_PKEY_fromdata().
+  */
+ #ifndef B_ENDIAN
+ # error "We haven't prepared little endian arrays"
+ #endif
+ const unsigned char priv[] = {
+     0xb9, 0x2f, 0x3c, 0xe6, 0x2f, 0xfb, 0x45, 0x68,
+     0x39, 0x96, 0xf0, 0x2a, 0xaf, 0x6c, 0xda, 0xf2,
+     0x89, 0x8a, 0x27, 0xbf, 0x39, 0x9b, 0x7e, 0x54,
+     0x21, 0xc2, 0xa1, 0xe5, 0x36, 0x12, 0x48, 0x5d
+ };
+ const unsigned char pub[] = {
+     0x04, 0xcf, 0x20, 0xfb, 0x9a, 0x1d, 0x11, 0x6c,
+     0x5e, 0x9f, 0xec, 0x38, 0x87, 0x6c, 0x1d, 0x2f,
+     0x58, 0x47, 0xab, 0xa3, 0x9b, 0x79, 0x23, 0xe6,
+     0xeb, 0x94, 0x6f, 0x97, 0xdb, 0xa3, 0x7d, 0xbd,
+     0xe5, 0x26, 0xca, 0x07, 0x17, 0x8d, 0x26, 0x75,
+     0xff, 0xcb, 0x8e, 0xb6, 0x84, 0xd0, 0x24, 0x02,
+     0x25, 0x8f, 0xb9, 0x33, 0x6e, 0xcf, 0x12, 0x16,
+     0x2f, 0x5c, 0xcd, 0x86, 0x71, 0xa8, 0xbf, 0x1a,
+     0x47
+ };
+ const OSSL_PARAM params[] = {
+     OSSL_PARAM_utf8_string("curve-name", "prime256v1"),
+     OSSL_PARAM_BN("priv", priv, sizeof(priv));
+     OSSL_PARAM_BN("pub", pub, sizeof(pub));
+     OSSL_PARAM_END
+ };
+
+ int main()
+ {
+     EVP_PKEY_CTX *ctx = EVP_PKEY_CTX_new_from_name(NULL, "EC", NULL);
+     EVP_PKEY *pkey = NULL;
+
+     if (ctx == NULL
+         || !EVP_PKEY_key_fromdata_init(ctx)
+         || !EVP_PKEY_fromdata(ctx, &pkey, params))
+         exit(1);
```

Comment:
again, I did understand Andy in this point.  I took his idea and made it mine.

Reply:
and once, again, when I contribute to linux I use a totally different style
and follow that as pedantically as you know me already.
Also gcc and (i am still learning gdb) have their own also special style.
Which has nothing to do with the way I would write my programs
when I start a totally new project.

Reply:
The extra indent of `if` condition expressions isn't mandated in our style.  We *allow it*, but the idea is so far ad-hoc, and has been agreed upon, because issues.